Introduction
------------

Here, we will create a single cell reference from our [published
work](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE198896) and
from single-cell RNAseq from two additional patients with Ewing sarcoma.

Load packages
-------------

    .libPaths("/ix1/acillo/arc85/packages/Rlibs_sarc028")

    suppressMessages({
    library(Seurat)
    library(tidyverse)
    library(patchwork)
    library(here)
    })

Load previously published data
------------------------------

    ewing_pub <- readRDS("../../01_input/cillo_et_al_ccr_2022_ewing_til.rds")

    table(ewing_pub@meta.data$tissue,ewing_pub@meta.data$sample_type)

    ##      
    ##       Ewing
    ##   TIL  5194

Read in new Ewing data
----------------------

    ewing_path <- "../../01_input/ewing_scrnaseq_new/"
    sample1 <- "LAF5160A88/filtered_feature_bc_matrix"
    sample2 <- "LAF5160A89/filtered_feature_bc_matrix"
    ewing_samples <- c(paste(ewing_path,sample1,sep=""),paste(ewing_path,sample2,sep=""))

    ewing_dat <- vector("list",length=2)

    for (i in 1:2) {
        ewing_raw <- Read10X(ewing_samples[i])
        ewing_dat[[i]] <- CreateSeuratObject(ewing_raw)
        print(paste("Finishing ",i," of ",length(ewing_samples),sep=""))
    }

    ## [1] "Finishing 1 of 2"
    ## [1] "Finishing 2 of 2"

    ewing_dat[[1]]@meta.data$sample <- "LAF5160A88"
    ewing_dat[[2]]@meta.data$sample <- "LAF5160A89"

    ewing_new <- merge(ewing_dat[[1]],ewing_dat[[2]])

    ## Warning in CheckDuplicateCellNames(object.list = objects): Some cell names are
    ## duplicated across objects provided. Renaming to enforce unique cell names.

Combine datasets
----------------

    ## Filter to shared genes
    DefaultAssay(ewing_pub) <- "RNA"

    set1 <- rownames(ewing_new)
    set2 <- rownames(ewing_pub)

    genes_use <- intersect(set1,set2)

    ewing_pub <- ewing_pub[genes_use,]

    ewing_new <- ewing_new[genes_use,]

    ## Combine data 
    ewing_comb <- merge(ewing_pub,ewing_new)

    DefaultAssay(ewing_comb) <- "RNA"

    ewing_comb@meta.data$scrnaseq_set <- ifelse(grepl("LAF",ewing_comb@meta.data$sample),"set2","set1")

    table(ewing_comb@meta.data$scrnaseq_set)

    ## 
    ##  set1  set2 
    ##  5194 12416

Filter based on gene features
-----------------------------

    mito_genes <- grep(pattern = "^MT-", x = rownames(x = ewing_comb), value = TRUE)
    percent_mito <- Matrix::colSums(GetAssayData(ewing_comb,slot="counts",assay="RNA")[mito_genes,])/Matrix::colSums(GetAssayData(ewing_comb,slot="counts",assay="RNA"))

    ewing_comb@meta.data$percent_mito <- percent_mito

    VlnPlot(ewing_comb,features="percent_mito")

![](01_integrate_two_refs_ewing-exported_files/figure-markdown_strict/unnamed-chunk-5-1.png)

    ewing_comb <- ewing_comb[,ewing_comb@meta.data$percent_mito<0.25]

    VlnPlot(ewing_comb,features="nFeature_RNA") +
        geom_hline(yintercept=200)

![](01_integrate_two_refs_ewing-exported_files/figure-markdown_strict/unnamed-chunk-5-2.png)

    ewing_comb <- ewing_comb[,ewing_comb@meta.data$nFeature_RNA>200]

    VlnPlot(ewing_comb,features="HBA2",slot="counts") +
        geom_hline(yintercept=5)

![](01_integrate_two_refs_ewing-exported_files/figure-markdown_strict/unnamed-chunk-5-3.png)

    ewing_comb <- ewing_comb[,GetAssayData(ewing_comb,assa="RNA",slot="counts")["HBA2",]<5]

Data integration
----------------

We need to perform data integration to account for differences in the
tumor populations from the two Ewing patients.

    ewing_list <- SplitObject(ewing_comb, split.by = "sample")
    ewing_list <- lapply(X = ewing_list, FUN = SCTransform)

    ## Calculating cell attributes from input UMI matrix: log_umi

    ## Variance stabilizing transformation of count matrix of size 10273 by 666

    ## Model formula is y ~ log_umi

    ## Get Negative Binomial regression parameters per gene

    ## Using 2000 genes, 666 cells

    ##   |                                                                              |                                                                      |   0%  |                                                                              |==================                                                    |  25%  |                                                                              |===================================                                   |  50%  |                                                                              |====================================================                  |  75%  |                                                                              |======================================================================| 100%

    ## Found 120 outliers - those will be ignored in fitting/regularization step

    ## Second step: Get residuals using fitted parameters for 10273 genes

    ##   |                                                                              |                                                                      |   0%  |                                                                              |===                                                                   |   5%  |                                                                              |=======                                                               |  10%  |                                                                              |==========                                                            |  14%  |                                                                              |=============                                                         |  19%  |                                                                              |=================                                                     |  24%  |                                                                              |====================                                                  |  29%  |                                                                              |=======================                                               |  33%  |                                                                              |===========================                                           |  38%  |                                                                              |==============================                                        |  43%  |                                                                              |=================================                                     |  48%  |                                                                              |=====================================                                 |  52%  |                                                                              |========================================                              |  57%  |                                                                              |===========================================                           |  62%  |                                                                              |===============================================                       |  67%  |                                                                              |==================================================                    |  71%  |                                                                              |=====================================================                 |  76%  |                                                                              |=========================================================             |  81%  |                                                                              |============================================================          |  86%  |                                                                              |===============================================================       |  90%  |                                                                              |===================================================================   |  95%  |                                                                              |======================================================================| 100%

    ## Computing corrected count matrix for 10273 genes

    ##   |                                                                              |                                                                      |   0%  |                                                                              |===                                                                   |   5%  |                                                                              |=======                                                               |  10%  |                                                                              |==========                                                            |  14%  |                                                                              |=============                                                         |  19%  |                                                                              |=================                                                     |  24%  |                                                                              |====================                                                  |  29%  |                                                                              |=======================                                               |  33%  |                                                                              |===========================                                           |  38%  |                                                                              |==============================                                        |  43%  |                                                                              |=================================                                     |  48%  |                                                                              |=====================================                                 |  52%  |                                                                              |========================================                              |  57%  |                                                                              |===========================================                           |  62%  |                                                                              |===============================================                       |  67%  |                                                                              |==================================================                    |  71%  |                                                                              |=====================================================                 |  76%  |                                                                              |=========================================================             |  81%  |                                                                              |============================================================          |  86%  |                                                                              |===============================================================       |  90%  |                                                                              |===================================================================   |  95%  |                                                                              |======================================================================| 100%

    ## Calculating gene attributes

    ## Wall clock passed: Time difference of 18.73443 secs

    ## Determine variable features

    ## Place corrected count matrix in counts slot

    ## Centering data matrix

    ## Set default assay to SCT

    ## Calculating cell attributes from input UMI matrix: log_umi

    ## Variance stabilizing transformation of count matrix of size 12836 by 2125

    ## Model formula is y ~ log_umi

    ## Get Negative Binomial regression parameters per gene

    ## Using 2000 genes, 2125 cells

    ##   |                                                                              |                                                                      |   0%  |                                                                              |==================                                                    |  25%  |                                                                              |===================================                                   |  50%  |                                                                              |====================================================                  |  75%  |                                                                              |======================================================================| 100%

    ## Found 121 outliers - those will be ignored in fitting/regularization step

    ## Second step: Get residuals using fitted parameters for 12836 genes

    ##   |                                                                              |                                                                      |   0%  |                                                                              |===                                                                   |   4%  |                                                                              |=====                                                                 |   8%  |                                                                              |========                                                              |  12%  |                                                                              |===========                                                           |  15%  |                                                                              |=============                                                         |  19%  |                                                                              |================                                                      |  23%  |                                                                              |===================                                                   |  27%  |                                                                              |======================                                                |  31%  |                                                                              |========================                                              |  35%  |                                                                              |===========================                                           |  38%  |                                                                              |==============================                                        |  42%  |                                                                              |================================                                      |  46%  |                                                                              |===================================                                   |  50%  |                                                                              |======================================                                |  54%  |                                                                              |========================================                              |  58%  |                                                                              |===========================================                           |  62%  |                                                                              |==============================================                        |  65%  |                                                                              |================================================                      |  69%  |                                                                              |===================================================                   |  73%  |                                                                              |======================================================                |  77%  |                                                                              |=========================================================             |  81%  |                                                                              |===========================================================           |  85%  |                                                                              |==============================================================        |  88%  |                                                                              |=================================================================     |  92%  |                                                                              |===================================================================   |  96%  |                                                                              |======================================================================| 100%

    ## Computing corrected count matrix for 12836 genes

    ##   |                                                                              |                                                                      |   0%  |                                                                              |===                                                                   |   4%  |                                                                              |=====                                                                 |   8%  |                                                                              |========                                                              |  12%  |                                                                              |===========                                                           |  15%  |                                                                              |=============                                                         |  19%  |                                                                              |================                                                      |  23%  |                                                                              |===================                                                   |  27%  |                                                                              |======================                                                |  31%  |                                                                              |========================                                              |  35%  |                                                                              |===========================                                           |  38%  |                                                                              |==============================                                        |  42%  |                                                                              |================================                                      |  46%  |                                                                              |===================================                                   |  50%  |                                                                              |======================================                                |  54%  |                                                                              |========================================                              |  58%  |                                                                              |===========================================                           |  62%  |                                                                              |==============================================                        |  65%  |                                                                              |================================================                      |  69%  |                                                                              |===================================================                   |  73%  |                                                                              |======================================================                |  77%  |                                                                              |=========================================================             |  81%  |                                                                              |===========================================================           |  85%  |                                                                              |==============================================================        |  88%  |                                                                              |=================================================================     |  92%  |                                                                              |===================================================================   |  96%  |                                                                              |======================================================================| 100%

    ## Calculating gene attributes

    ## Wall clock passed: Time difference of 52.7718 secs

    ## Determine variable features

    ## Place corrected count matrix in counts slot

    ## Centering data matrix

    ## Set default assay to SCT

    ## Calculating cell attributes from input UMI matrix: log_umi

    ## Variance stabilizing transformation of count matrix of size 11845 by 2362

    ## Model formula is y ~ log_umi

    ## Get Negative Binomial regression parameters per gene

    ## Using 2000 genes, 2362 cells

    ##   |                                                                              |                                                                      |   0%  |                                                                              |==================                                                    |  25%  |                                                                              |===================================                                   |  50%  |                                                                              |====================================================                  |  75%  |                                                                              |======================================================================| 100%

    ## Found 128 outliers - those will be ignored in fitting/regularization step

    ## Second step: Get residuals using fitted parameters for 11845 genes

    ##   |                                                                              |                                                                      |   0%  |                                                                              |===                                                                   |   4%  |                                                                              |======                                                                |   8%  |                                                                              |=========                                                             |  12%  |                                                                              |============                                                          |  17%  |                                                                              |===============                                                       |  21%  |                                                                              |==================                                                    |  25%  |                                                                              |====================                                                  |  29%  |                                                                              |=======================                                               |  33%  |                                                                              |==========================                                            |  38%  |                                                                              |=============================                                         |  42%  |                                                                              |================================                                      |  46%  |                                                                              |===================================                                   |  50%  |                                                                              |======================================                                |  54%  |                                                                              |=========================================                             |  58%  |                                                                              |============================================                          |  62%  |                                                                              |===============================================                       |  67%  |                                                                              |==================================================                    |  71%  |                                                                              |====================================================                  |  75%  |                                                                              |=======================================================               |  79%  |                                                                              |==========================================================            |  83%  |                                                                              |=============================================================         |  88%  |                                                                              |================================================================      |  92%  |                                                                              |===================================================================   |  96%  |                                                                              |======================================================================| 100%

    ## Computing corrected count matrix for 11845 genes

    ##   |                                                                              |                                                                      |   0%  |                                                                              |===                                                                   |   4%  |                                                                              |======                                                                |   8%  |                                                                              |=========                                                             |  12%  |                                                                              |============                                                          |  17%  |                                                                              |===============                                                       |  21%  |                                                                              |==================                                                    |  25%  |                                                                              |====================                                                  |  29%  |                                                                              |=======================                                               |  33%  |                                                                              |==========================                                            |  38%  |                                                                              |=============================                                         |  42%  |                                                                              |================================                                      |  46%  |                                                                              |===================================                                   |  50%  |                                                                              |======================================                                |  54%  |                                                                              |=========================================                             |  58%  |                                                                              |============================================                          |  62%  |                                                                              |===============================================                       |  67%  |                                                                              |==================================================                    |  71%  |                                                                              |====================================================                  |  75%  |                                                                              |=======================================================               |  79%  |                                                                              |==========================================================            |  83%  |                                                                              |=============================================================         |  88%  |                                                                              |================================================================      |  92%  |                                                                              |===================================================================   |  96%  |                                                                              |======================================================================| 100%

    ## Calculating gene attributes

    ## Wall clock passed: Time difference of 1.021176 mins

    ## Determine variable features

    ## Place corrected count matrix in counts slot

    ## Centering data matrix

    ## Set default assay to SCT

    ## Calculating cell attributes from input UMI matrix: log_umi

    ## Variance stabilizing transformation of count matrix of size 16345 by 3773

    ## Model formula is y ~ log_umi

    ## Get Negative Binomial regression parameters per gene

    ## Using 2000 genes, 3773 cells

    ##   |                                                                              |                                                                      |   0%  |                                                                              |==================                                                    |  25%  |                                                                              |===================================                                   |  50%  |                                                                              |====================================================                  |  75%  |                                                                              |======================================================================| 100%

    ## Found 81 outliers - those will be ignored in fitting/regularization step

    ## Second step: Get residuals using fitted parameters for 16345 genes

    ##   |                                                                              |                                                                      |   0%  |                                                                              |==                                                                    |   3%  |                                                                              |====                                                                  |   6%  |                                                                              |======                                                                |   9%  |                                                                              |========                                                              |  12%  |                                                                              |===========                                                           |  15%  |                                                                              |=============                                                         |  18%  |                                                                              |===============                                                       |  21%  |                                                                              |=================                                                     |  24%  |                                                                              |===================                                                   |  27%  |                                                                              |=====================                                                 |  30%  |                                                                              |=======================                                               |  33%  |                                                                              |=========================                                             |  36%  |                                                                              |============================                                          |  39%  |                                                                              |==============================                                        |  42%  |                                                                              |================================                                      |  45%  |                                                                              |==================================                                    |  48%  |                                                                              |====================================                                  |  52%  |                                                                              |======================================                                |  55%  |                                                                              |========================================                              |  58%  |                                                                              |==========================================                            |  61%  |                                                                              |=============================================                         |  64%  |                                                                              |===============================================                       |  67%  |                                                                              |=================================================                     |  70%  |                                                                              |===================================================                   |  73%  |                                                                              |=====================================================                 |  76%  |                                                                              |=======================================================               |  79%  |                                                                              |=========================================================             |  82%  |                                                                              |===========================================================           |  85%  |                                                                              |==============================================================        |  88%  |                                                                              |================================================================      |  91%  |                                                                              |==================================================================    |  94%  |                                                                              |====================================================================  |  97%  |                                                                              |======================================================================| 100%

    ## Computing corrected count matrix for 16345 genes

    ##   |                                                                              |                                                                      |   0%  |                                                                              |==                                                                    |   3%  |                                                                              |====                                                                  |   6%  |                                                                              |======                                                                |   9%  |                                                                              |========                                                              |  12%  |                                                                              |===========                                                           |  15%  |                                                                              |=============                                                         |  18%  |                                                                              |===============                                                       |  21%  |                                                                              |=================                                                     |  24%  |                                                                              |===================                                                   |  27%  |                                                                              |=====================                                                 |  30%  |                                                                              |=======================                                               |  33%  |                                                                              |=========================                                             |  36%  |                                                                              |============================                                          |  39%  |                                                                              |==============================                                        |  42%  |                                                                              |================================                                      |  45%  |                                                                              |==================================                                    |  48%  |                                                                              |====================================                                  |  52%  |                                                                              |======================================                                |  55%  |                                                                              |========================================                              |  58%  |                                                                              |==========================================                            |  61%  |                                                                              |=============================================                         |  64%  |                                                                              |===============================================                       |  67%  |                                                                              |=================================================                     |  70%  |                                                                              |===================================================                   |  73%  |                                                                              |=====================================================                 |  76%  |                                                                              |=======================================================               |  79%  |                                                                              |=========================================================             |  82%  |                                                                              |===========================================================           |  85%  |                                                                              |==============================================================        |  88%  |                                                                              |================================================================      |  91%  |                                                                              |==================================================================    |  94%  |                                                                              |====================================================================  |  97%  |                                                                              |======================================================================| 100%

    ## Calculating gene attributes

    ## Wall clock passed: Time difference of 1.61299 mins

    ## Determine variable features

    ## Place corrected count matrix in counts slot

    ## Centering data matrix

    ## Set default assay to SCT

    ## Calculating cell attributes from input UMI matrix: log_umi

    ## Variance stabilizing transformation of count matrix of size 17075 by 4254

    ## Model formula is y ~ log_umi

    ## Get Negative Binomial regression parameters per gene

    ## Using 2000 genes, 4254 cells

    ##   |                                                                              |                                                                      |   0%  |                                                                              |==================                                                    |  25%  |                                                                              |===================================                                   |  50%  |                                                                              |====================================================                  |  75%  |                                                                              |======================================================================| 100%

    ## Found 99 outliers - those will be ignored in fitting/regularization step

    ## Second step: Get residuals using fitted parameters for 17075 genes

    ##   |                                                                              |                                                                      |   0%  |                                                                              |==                                                                    |   3%  |                                                                              |====                                                                  |   6%  |                                                                              |======                                                                |   9%  |                                                                              |========                                                              |  11%  |                                                                              |==========                                                            |  14%  |                                                                              |============                                                          |  17%  |                                                                              |==============                                                        |  20%  |                                                                              |================                                                      |  23%  |                                                                              |==================                                                    |  26%  |                                                                              |====================                                                  |  29%  |                                                                              |======================                                                |  31%  |                                                                              |========================                                              |  34%  |                                                                              |==========================                                            |  37%  |                                                                              |============================                                          |  40%  |                                                                              |==============================                                        |  43%  |                                                                              |================================                                      |  46%  |                                                                              |==================================                                    |  49%  |                                                                              |====================================                                  |  51%  |                                                                              |======================================                                |  54%  |                                                                              |========================================                              |  57%  |                                                                              |==========================================                            |  60%  |                                                                              |============================================                          |  63%  |                                                                              |==============================================                        |  66%  |                                                                              |================================================                      |  69%  |                                                                              |==================================================                    |  71%  |                                                                              |====================================================                  |  74%  |                                                                              |======================================================                |  77%  |                                                                              |========================================================              |  80%  |                                                                              |==========================================================            |  83%  |                                                                              |============================================================          |  86%  |                                                                              |==============================================================        |  89%  |                                                                              |================================================================      |  91%  |                                                                              |==================================================================    |  94%  |                                                                              |====================================================================  |  97%  |                                                                              |======================================================================| 100%

    ## Computing corrected count matrix for 17075 genes

    ##   |                                                                              |                                                                      |   0%  |                                                                              |==                                                                    |   3%  |                                                                              |====                                                                  |   6%  |                                                                              |======                                                                |   9%  |                                                                              |========                                                              |  11%  |                                                                              |==========                                                            |  14%  |                                                                              |============                                                          |  17%  |                                                                              |==============                                                        |  20%  |                                                                              |================                                                      |  23%  |                                                                              |==================                                                    |  26%  |                                                                              |====================                                                  |  29%  |                                                                              |======================                                                |  31%  |                                                                              |========================                                              |  34%  |                                                                              |==========================                                            |  37%  |                                                                              |============================                                          |  40%  |                                                                              |==============================                                        |  43%  |                                                                              |================================                                      |  46%  |                                                                              |==================================                                    |  49%  |                                                                              |====================================                                  |  51%  |                                                                              |======================================                                |  54%  |                                                                              |========================================                              |  57%  |                                                                              |==========================================                            |  60%  |                                                                              |============================================                          |  63%  |                                                                              |==============================================                        |  66%  |                                                                              |================================================                      |  69%  |                                                                              |==================================================                    |  71%  |                                                                              |====================================================                  |  74%  |                                                                              |======================================================                |  77%  |                                                                              |========================================================              |  80%  |                                                                              |==========================================================            |  83%  |                                                                              |============================================================          |  86%  |                                                                              |==============================================================        |  89%  |                                                                              |================================================================      |  91%  |                                                                              |==================================================================    |  94%  |                                                                              |====================================================================  |  97%  |                                                                              |======================================================================| 100%

    ## Calculating gene attributes

    ## Wall clock passed: Time difference of 1.928001 mins

    ## Determine variable features

    ## Place corrected count matrix in counts slot

    ## Centering data matrix

    ## Set default assay to SCT

    features <- SelectIntegrationFeatures(object.list = ewing_list, nfeatures = 2000)
    ewing_list <- PrepSCTIntegration(object.list = ewing_list, anchor.features = features)

    features <- SelectIntegrationFeatures(object.list = ewing_list, nfeatures = 2000)

    ewing_anchors <- FindIntegrationAnchors(object.list = ewing_list,
                                            normalization.method = "SCT",
                                            anchor.features = features
                                            )

    ## Finding all pairwise anchors

    ## Running CCA

    ## Merging objects

    ## Finding neighborhoods

    ## Finding anchors

    ##  Found 2544 anchors

    ## Filtering anchors

    ##  Retained 2038 anchors

    ## Running CCA

    ## Merging objects

    ## Finding neighborhoods

    ## Finding anchors

    ##  Found 2475 anchors

    ## Filtering anchors

    ##  Retained 1976 anchors

    ## Running CCA

    ## Merging objects

    ## Finding neighborhoods

    ## Finding anchors

    ##  Found 4887 anchors

    ## Filtering anchors

    ##  Retained 3471 anchors

    ## Running CCA

    ## Merging objects

    ## Finding neighborhoods

    ## Finding anchors

    ##  Found 2820 anchors

    ## Filtering anchors

    ##  Retained 1987 anchors

    ## Running CCA

    ## Merging objects

    ## Finding neighborhoods

    ## Finding anchors

    ##  Found 5543 anchors

    ## Filtering anchors

    ##  Retained 3784 anchors

    ## Running CCA

    ## Merging objects

    ## Finding neighborhoods

    ## Finding anchors

    ##  Found 5518 anchors

    ## Filtering anchors

    ##  Retained 4131 anchors

    ## Running CCA

    ## Merging objects

    ## Finding neighborhoods

    ## Finding anchors

    ##  Found 2913 anchors

    ## Filtering anchors

    ##  Retained 1533 anchors

    ## Running CCA

    ## Merging objects

    ## Finding neighborhoods

    ## Finding anchors

    ##  Found 6676 anchors

    ## Filtering anchors

    ##  Retained 3630 anchors

    ## Running CCA

    ## Merging objects

    ## Finding neighborhoods

    ## Finding anchors

    ##  Found 7132 anchors

    ## Filtering anchors

    ##  Retained 3785 anchors

    ## Running CCA

    ## Merging objects

    ## Finding neighborhoods

    ## Finding anchors

    ##  Found 7012 anchors

    ## Filtering anchors

    ##  Retained 4362 anchors

    ewing_comb_sct <- IntegrateData(anchorset = ewing_anchors, normalization.method = "SCT")

    ## Merging dataset 1 into 2

    ## Extracting anchors for merged samples

    ## Finding integration vectors

    ## Finding integration vector weights

    ## Integrating data

    ## Merging dataset 2 1 into 4

    ## Warning: Attempting to merge an SCTAssay with another Assay type 
    ## Converting all to standard Assay objects.

    ## Extracting anchors for merged samples

    ## Warning: Attempting to merge an SCTAssay with another Assay type 
    ## Converting all to standard Assay objects.

    ## Finding integration vectors

    ## Finding integration vector weights

    ## Integrating data

    ## Merging dataset 3 into 4 2 1

    ## Extracting anchors for merged samples

    ## Finding integration vectors

    ## Finding integration vector weights

    ## Integrating data

    ## Merging dataset 5 into 4 2 1 3

    ## Extracting anchors for merged samples

    ## Finding integration vectors

    ## Finding integration vector weights

    ## Integrating data

    ewing_comb_sct <- RunPCA(ewing_comb_sct, verbose = FALSE)

    ElbowPlot(ewing_comb_sct,ndims=50)

![](01_integrate_two_refs_ewing-exported_files/figure-markdown_strict/unnamed-chunk-6-1.png)

    ewing_comb_sct <- RunUMAP(ewing_comb_sct, reduction = "pca", dims = 1:30)

    ## Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
    ## To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'
    ## This message will be shown once per session

    ## 09:07:41 UMAP embedding parameters a = 0.9922 b = 1.112

    ## 09:07:41 Read 13180 rows and found 30 numeric columns

    ## 09:07:41 Using Annoy for neighbor search, n_neighbors = 30

    ## 09:07:41 Building Annoy index with metric = cosine, n_trees = 50

    ## 0%   10   20   30   40   50   60   70   80   90   100%

    ## [----|----|----|----|----|----|----|----|----|----|

    ## **************************************************|
    ## 09:07:44 Writing NN index file to temp file /tmp/RtmpF4yNA8/file40723b898bb7
    ## 09:07:44 Searching Annoy index using 1 thread, search_k = 3000
    ## 09:07:53 Annoy recall = 100%
    ## 09:07:54 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 30
    ## 09:07:56 Initializing from normalized Laplacian + noise (using irlba)
    ## 09:07:57 Commencing optimization for 200 epochs, with 591286 positive edges
    ## 09:08:24 Optimization finished

    ewing_comb_sct <- FindNeighbors(ewing_comb_sct, reduction = "pca", dims = 1:30)

    ## Computing nearest neighbor graph
    ## Computing SNN

    ewing_comb_sct <- FindClusters(ewing_comb_sct, resolution = c(0.3,0.5,0.7,1))

    ## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
    ## 
    ## Number of nodes: 13180
    ## Number of edges: 503587
    ## 
    ## Running Louvain algorithm...
    ## Maximum modularity in 10 random starts: 0.9308
    ## Number of communities: 17
    ## Elapsed time: 4 seconds
    ## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
    ## 
    ## Number of nodes: 13180
    ## Number of edges: 503587
    ## 
    ## Running Louvain algorithm...
    ## Maximum modularity in 10 random starts: 0.9143
    ## Number of communities: 22
    ## Elapsed time: 4 seconds
    ## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
    ## 
    ## Number of nodes: 13180
    ## Number of edges: 503587
    ## 
    ## Running Louvain algorithm...
    ## Maximum modularity in 10 random starts: 0.9006
    ## Number of communities: 25
    ## Elapsed time: 4 seconds
    ## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
    ## 
    ## Number of nodes: 13180
    ## Number of edges: 503587
    ## 
    ## Running Louvain algorithm...
    ## Maximum modularity in 10 random starts: 0.8835
    ## Number of communities: 27
    ## Elapsed time: 4 seconds

    p1 <- DimPlot(ewing_comb_sct, reduction = "umap", group.by = "sample")
    p2 <- DimPlot(ewing_comb_sct, reduction = "umap", group.by="integrated_snn_res.0.5",label = TRUE, repel = TRUE)
    p1 + p2

![](01_integrate_two_refs_ewing-exported_files/figure-markdown_strict/unnamed-chunk-6-2.png)

    DefaultAssay(ewing_comb_sct) <- "RNA"

    ewing_comb_sct <- NormalizeData(ewing_comb_sct)

    table(ewing_comb_sct@meta.data$cell_types,ewing_comb_sct@meta.data$integrated_snn_res.0.5) %>%
        pheatmap::pheatmap(.,scale="row")

![](01_integrate_two_refs_ewing-exported_files/figure-markdown_strict/unnamed-chunk-6-3.png)

    p3 <- DimPlot(ewing_comb_sct, reduction = "umap", group.by = "cell_types")
    p4 <- DimPlot(ewing_comb_sct, reduction = "umap", group.by="integrated_snn_res.0.5",label = TRUE, repel = TRUE)
    p5 <- FeaturePlot(ewing_comb_sct,c("CD3D","CD8A","CD14","PTPRC"))

    p3 + p4 + p5

![](01_integrate_two_refs_ewing-exported_files/figure-markdown_strict/unnamed-chunk-6-4.png)

    FeaturePlot(ewing_comb_sct,c("EPCAM","PTPRC","CD99","FLI1","CAV1","MMP9","COL1A1"))

![](01_integrate_two_refs_ewing-exported_files/figure-markdown_strict/unnamed-chunk-6-5.png)

    metadata <- ewing_comb_sct@meta.data

    Idents(ewing_comb_sct) <- "integrated_snn_res.0.5"

    metadata <- metadata %>%
        mutate(crude_cell_types=recode(integrated_snn_res.0.5,
                                            `0`="Tumor cells 1",
                                            `1`="T cells",
                                            `2`="CD14+CD16+ Macrophage",
                                            `3`="Tumor cells 2",
                                            `4`="CD14+CD16- Monocytes",
                                            `5`="Tumor cells 3",
                                            `6`="T cells",
                                            `7`="B cells",
                                            `8`="Tumor cells 4",
                                            `9`="Tumor cells 5",
                                            `10`="CD1C+ DCs",
                                            `11`="Fibroblasts",
                                            `12`="CD14+CD16+ Monocytes",
                                            `13`="Tumor cells 6",
                                            `14`="NK cells",
                                            `15`="Tumor cells 7",
                                            `16`="CD14+CD16+ Macrophage",
                                            `17`="Osteoclasts",
                                            `18`="pDCs",
                                            `19`="Tumor cells 8",
                                            `20`="CD1C+ DCs",
                                            `21`="Plasmablasts"
                                       )
               )

    metadata$scrnaseq_crude_cell_types <- as.character(metadata$crude_cell_types)

    ewing_comb_sct@meta.data$scrnaseq_crude_cell_types <- metadata$scrnaseq_crude_cell_types

    DimPlot(ewing_comb_sct,group.by="scrnaseq_crude_cell_types",label=T,repel=T)

![](01_integrate_two_refs_ewing-exported_files/figure-markdown_strict/unnamed-chunk-6-6.png)

Save results and session info
-----------------------------

    # saveRDS(ewing_comb_sct,file="../../03_output/Ewing_scRNAseq_reference.rds")

    sessionInfo()

    ## R version 4.2.0 (2022-04-22)
    ## Platform: x86_64-pc-linux-gnu (64-bit)
    ## Running under: Red Hat Enterprise Linux
    ## 
    ## Matrix products: default
    ## BLAS:   /usr/lib64/libblas.so.3.4.2
    ## LAPACK: /usr/lib64/liblapack.so.3.4.2
    ## 
    ## locale:
    ## [1] en_US.UTF-8
    ## 
    ## attached base packages:
    ## [1] stats     graphics  grDevices utils     datasets  methods   base     
    ## 
    ## other attached packages:
    ##  [1] here_1.0.1         patchwork_1.1.2    lubridate_1.9.2    forcats_1.0.0     
    ##  [5] stringr_1.4.1      dplyr_1.1.0        purrr_1.0.1        readr_2.1.4       
    ##  [9] tidyr_1.3.0        tibble_3.1.8       ggplot2_3.4.1      tidyverse_2.0.0   
    ## [13] SeuratObject_4.1.3 Seurat_4.3.0.1    
    ## 
    ## loaded via a namespace (and not attached):
    ##   [1] ggbeeswarm_0.6.0       Rtsne_0.16             colorspace_2.0-3      
    ##   [4] deldir_1.0-6           ellipsis_0.3.2         ggridges_0.5.4        
    ##   [7] rprojroot_2.0.3        spatstat.data_3.0-0    farver_2.1.1          
    ##  [10] leiden_0.4.3           listenv_0.8.0          ggrepel_0.9.2         
    ##  [13] fansi_1.0.3            R.methodsS3_1.8.2      codetools_0.2-18      
    ##  [16] splines_4.2.0          knitr_1.39             polyclip_1.10-4       
    ##  [19] jsonlite_1.8.4         ica_1.0-3              cluster_2.1.3         
    ##  [22] R.oo_1.25.0            png_0.1-7              pheatmap_1.0.12       
    ##  [25] uwot_0.1.14            shiny_1.7.3            sctransform_0.3.5     
    ##  [28] spatstat.sparse_3.0-0  compiler_4.2.0         httr_1.4.4            
    ##  [31] Matrix_1.5-3           fastmap_1.1.0          lazyeval_0.2.2        
    ##  [34] cli_3.4.1              later_1.3.0            htmltools_0.5.6.1     
    ##  [37] tools_4.2.0            igraph_1.3.5           gtable_0.3.1          
    ##  [40] glue_1.6.2             RANN_2.6.1             reshape2_1.4.4        
    ##  [43] Rcpp_1.0.9             scattermore_0.8        vctrs_0.5.2           
    ##  [46] spatstat.explore_3.0-5 nlme_3.1-157           progressr_0.11.0      
    ##  [49] lmtest_0.9-40          spatstat.random_3.0-1  xfun_0.40             
    ##  [52] globals_0.16.2         timechange_0.2.0       mime_0.12             
    ##  [55] miniUI_0.1.1.1         lifecycle_1.0.3        irlba_2.3.5.1         
    ##  [58] goftest_1.2-3          future_1.29.0          MASS_7.3-57           
    ##  [61] zoo_1.8-11             scales_1.3.0           hms_1.1.2             
    ##  [64] promises_1.2.0.1       spatstat.utils_3.0-1   parallel_4.2.0        
    ##  [67] RColorBrewer_1.1-3     yaml_2.3.6             reticulate_1.26       
    ##  [70] pbapply_1.6-0          gridExtra_2.3          ggrastr_1.0.1         
    ##  [73] stringi_1.7.8          highr_0.9              rlang_1.1.1           
    ##  [76] pkgconfig_2.0.3        matrixStats_0.63.0     evaluate_0.15         
    ##  [79] lattice_0.20-45        ROCR_1.0-11            tensor_1.5            
    ##  [82] labeling_0.4.2         htmlwidgets_1.5.4      cowplot_1.1.1         
    ##  [85] tidyselect_1.2.0       parallelly_1.32.1      RcppAnnoy_0.0.20      
    ##  [88] plyr_1.8.8             magrittr_2.0.3         R6_2.5.1              
    ##  [91] generics_0.1.3         pillar_1.8.1           withr_2.5.0           
    ##  [94] fitdistrplus_1.1-8     survival_3.5-3         abind_1.4-5           
    ##  [97] sp_1.5-1               future.apply_1.10.0    KernSmooth_2.23-20    
    ## [100] utf8_1.2.2             spatstat.geom_3.0-3    plotly_4.10.1         
    ## [103] tzdb_0.3.0             rmarkdown_2.25         grid_4.2.0            
    ## [106] data.table_1.14.6      digest_0.6.30          xtable_1.8-4          
    ## [109] httpuv_1.6.6           R.utils_2.12.1         munsell_0.5.0         
    ## [112] beeswarm_0.4.0         viridisLite_0.4.1      vipor_0.4.5
